### DISCUSSION TITLE
How to change the temporary folder

### DISCUSSION THREAD

**Discussion by Blank-c** [pyinstaller/pyinstaller#6977] [Help] [answered]
I want to change the `_MEIXXXXX` name of the folder to something else, how can I?

**Comment by rokm** [comment, marked as answer]
The naming scheme is hardcoded in bootloader, so if you really want to change it, you'll need to modify `pyi_get_temp_path` function ([Windiws](https://github.com/pyinstaller/pyinstaller/blob/97e73438062f584645f508c120ede6624b38a228/bootloader/src/pyi_utils.c#L334) or [POSIX](https://github.com/pyinstaller/pyinstaller/blob/97e73438062f584645f508c120ede6624b38a228/bootloader/src/pyi_utils.c#L390) codepath) and [recompile bootloader](https://pyinstaller.org/en/stable/bootloader-building.html). This requires installing from sdist (or git repository checkout) instead of using PyPI wheels with pre-compiled bootloader.

**Reply by Blank-c** [reply]
Ye Ik that much, just wanted to ask about the location of the `pyi_get_temp_path` :D thanks

**Reply by Blank-c** [reply]
so from what I understood, does it create a tempfolder based on its process id??

**Reply by rokm** [reply]
> so from what I understood, does it create a tempfolder based on its process id??

Yes, on Windows, the process id is included in the tempdir basename. If there was collision due to pid recycling, the `_wtempnam` would probably append additional identifier to make it unique.
